lib/middleware/query.js

```diff
@@ -23,6 +24,7 @@ var qs = require('qs');
  */
 
 module.exports = function query(options) {
+  deprecate('default extended query parser is deprecated, set explicitly to override');
```

Comment:
This is still logged even when the user changes the query parser setting to something explicitly, though.

Reply:
yes, I had a doubt on that. I made it this way, because I was seeing that we are hard-coding the value. But agree, if user could override that.

Made the change to check the parser name before deprecating. Please take a look!